### ISSUE TITLE
Why add a dropout parameter to the generation of attention map 

### ISSUE THREAD

**Issue by Epiphqny** [facebookresearch/detr#220]
Thanks for your work, what's the purpose of dropout in attention map generation ?https://github.com/facebookresearch/detr/blob/5e66b4cd15b2b182da347103dd16578d28b49d69/models/segmentation.py#L34
The number is set to 0, have you tried to use it in the model ?

**Comment by fmassa** [comment]
Hi,

The default implementation of MultiHeadAttention uses dropout internally to combat overfitting, [see the original implementation in PyTorch](https://github.com/pytorch/pytorch/blob/24ca6aab024f4e45e462e387ffd710a1d25b59da/torch/nn/functional.py#L4134).

@alcinos can comment further if he tried changing the dropout of the segmentation head to something else than 0.

**Comment by Epiphqny** [comment]
Thanks for your answer, I will tune the parameter and see what happens.
